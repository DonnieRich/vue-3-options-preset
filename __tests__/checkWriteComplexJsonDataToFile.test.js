const { vol } = require("memfs");

jest.mock("fs/promises");

const { writeJsonDataToFile } = require("../src/bin/writeJsonDataToFile");

describe(writeJsonDataToFile, () => {
    beforeEach(() => {
        vol.reset();
    });

    it("It should pass if the file package.json has been updated with the data provided", async () => {

        const data = {
            check: true,
            dependencies: {
                first: "first",
                second: 2
            },
            devDependencies: {
                third: false,
                fourth: "ok"
            }
        };

        vol.fromJSON(
            {
                "/package.json": "..."
            },
            '/'
        );

        await writeJsonDataToFile("/package.json", data);
        expect(vol.toJSON()).toMatchSnapshot();
    });
});