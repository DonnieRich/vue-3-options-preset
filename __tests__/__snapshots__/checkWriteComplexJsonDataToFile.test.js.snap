// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeJsonDataToFile It should pass if the file package.json has been updated with the data provided 1`] = `
{
  "/package.json": "{"check":true,"dependencies":{"first":"first","second":2},"devDependencies":{"third":false,"fourth":"ok"}}",
}
`;
